File: dataset/Intermediate/Translation/Avatar/v1-Python-code/codeforces_462_B.py
import threading
import queue

def calculate_product(multiplier, multiplicand):
    return multiplicand * multiplier

def my_decorator(func):

    def dec_result(*args, **kwargs):
        res = func(*args, **kwargs)
        return res
    return dec_result
(n, k) = map(int, input().split())
input_string = input()
multiplicand = [0]
multiplier = 26
result_queue = queue.Queue()

def threaded_calculation(queue):
    result = calculate_product(multiplier, multiplicand)
    queue.put(result)
calculation_thread = threading.Thread(target=threaded_calculation, args=(result_queue,))
calculation_thread.start()
calculation_thread.join()
product_result = result_queue.get()
count = product_result
for c in input_string:
    count[ord(c) - ord('A')] += 1
count.sort(reverse=True)
res = 0

@my_decorator
def recursive_summation(i, stop, step):
    global k, res
    if step == 0 or (step > 0 and i >= stop) or (step < 0 and i <= stop):
        return
    if count[i] >= k:
        res += k * k
        print(res)
        exit()
    k -= count[i]
    res += count[i] ** 2
    recursive_summation(i + step, stop, step)
recursive_summation(0, 26, 1)
print(res)

File: scripts/GAs/Experiment_Results_v1-Python-code/intermediate/SR/Translation/use_test/WizardCoder-15B-V1.0/Avatar/Python/Java/codeforces_462_B.java
```java
import java.util.*;
import java.util.concurrent.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        String input_string = sc.next();
        int[] multiplicand = new int[26];
        int multiplier = 26;
        Queue<Integer> result_queue = new LinkedList<>();
        Thread calculation_thread = new Thread(() -> {
            int result = calculate_product(multiplier, multiplicand);
            result_queue.offer(result);
        });
        calculation_thread.start();
        try {
            calculation_thread.join();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        int product_result = result_queue.poll();
        int[] count = new int[26];
        for (char c : input_string.toCharArray()) {
            count[c - 'A']++;
        }
        Arrays.sort(count);
        int res = 0;
        recursive_summation(0, 26, 1, k, count, res);
        System.out.println(res);
    }

    public static int calculate_product(int multiplier, int[] multiplicand) {
        int product = 1;
        for (int i = 0; i < 26; i++) {
            product *= multiplicand[i];
        }
        return product;
    }

    public static void recursive_summation(int i, int stop, int step, int k, int[] count, int res) {
        if (step == 0 || (step > 0 && i >= stop) || (step < 0 && i <= stop)) {
            return;
        }
        if (count[i] >= k) {
            res += k * k;
            System.out.println(res);
            System.exit(0);
        }
        k -= count[i];
        res += count[i] * count[i];
        recursive_summation(i + step, stop, step, k, count, res);
    }
}
```
